ledger/cow.go

```diff
@@ -255,6 +279,9 @@ func (cb *roundCowState) commitToParent() {
 			}
 		}
 	}
+	for gidx, tc := range cb.trackedCreatables {
+		cb.commitParent.trackedCreatables[gidx] = tc
+	}
```

Comment:
this is not really needed. In eval:
```golang
cow := eval.state.child(len(txgroup))
...
for gi, txad := range txgroup {
	var txib transactions.SignedTxnInBlock
	cow.setGroupIdx(gi)
	err := eval.transaction(txad.SignedTxn, evalParams[gi], txad.ApplyData, cow, &txib)
...
}
```
So there is a single cow for entire group and tracking stuff should only exist within it and have not been propagated down to block's cow where it does not make sense.

Reply:
I thought so as well, but then I noticed that child cows are used during the StatefulEval function:
```go
func (cb *roundCowState) StatefulEval(params logic.EvalParams, aidx basics.AppIndex, program []byte) (pass bool, evalDelta basics.EvalDelta, err error) {
	// Make a child cow to eval our program in
	calf := cb.child(1)
	params.Ledger, err = newLogicLedger(calf, aidx)
	if err != nil {
		return false, basics.EvalDelta{}, err
	}
    ...
```
However, I agree with your premise and I'll play around and see if I can get it to work without committing state to the parent.

Reply:
this child cow is only for TEAL evaluation so it does not affect `trackedCreatables`. So copying them into child as already implemented should be sufficient: group cow has `trackedCreatables` filled, they are copying to a child in `StatefulEval`, logic ledger has all data needed. Data are read only and do not need to be copied back to group cow.

Reply:
Oh I see - I misunderstood. I'll update it.